app/file.go

```diff
@@ -593,6 +594,19 @@ func (a *App) UploadFileX(channelId, name string, input io.Reader,
 		return t.fileinfo, aerr
 	}
 
+	// attempt to extract width & height out of svg
+	if !t.Raw && t.fileinfo.IsImage() && t.fileinfo.MimeType == "image/svg+xml" {
+		svgInfo := model.GetSVGInfoForBytes(t.buf.Bytes())
+		var svgViewBox = strings.Split(svgInfo.ViewBox, " ")
+		if len(svgViewBox) == 4 {
+			t.fileinfo.Width, _ = strconv.Atoi(svgViewBox[2])
```

Comment:
I'm guessing that if we can't parse the file info size as an integer, the `nil` default of `0` is OK? Would we want any other behaviour? (e.g. don't set any of these bits?)